os/drivers/audio/alc5658.c

```diff
@@ -422,6 +481,12 @@ static int alc5658_getcaps(FAR struct audio_lowerhalf_s *dev, int type, FAR stru
 	DEBUGASSERT(caps && caps->ac_len >= sizeof(struct audio_caps_s));
 	audvdbg("type=%d ac_type=%d\n", type, caps->ac_type);
 
+#ifndef CONFIG_DEBUG
+	if (!caps) {
+		return -1;
```

Comment:
You may need to change this -1 to ERROR or FAIL ?

Reply:
I checked this part. Actually it returns a count during success. I.e. caps->ac_len. So I return negative len instead of FAIL and ERROR